pandas/tests/indexes/test_numeric.py

```diff
@@ -233,27 +226,25 @@ def test_astype(self):
 
         # GH 12881
         # a float astype int
-        for dtype in ['int16', 'int32', 'int64']:
-            i = Float64Index([0, 1, 2])
-            result = i.astype(dtype)
-            expected = Int64Index([0, 1, 2])
-            tm.assert_index_equal(result, expected)
-
-            i = Float64Index([0, 1.1, 2])
-            result = i.astype(dtype)
-            expected = Int64Index([0, 1, 2])
-            tm.assert_index_equal(result, expected)
-
-        for dtype in ['float32', 'float64']:
-            i = Float64Index([0, 1, 2])
-            result = i.astype(dtype)
-            expected = i
-            tm.assert_index_equal(result, expected)
-
-            i = Float64Index([0, 1.1, 2])
-            result = i.astype(dtype)
-            expected = Index(i.values.astype(dtype))
-            tm.assert_index_equal(result, expected)
+        i = Float64Index([0, 1, 2])
+        result = i.astype('int64')
```

Comment:
you can't change this. a ``float32`` should simply be promoted to ``float64`` (and likewise ``int32`` to ``int64``)